ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerFragment.kt

```diff
@@ -68,11 +68,27 @@ class HomeScreenMapContainerFragment : AbstractMapContainerFragment() {
       .`as`(RxAutoDispose.autoDisposable(this))
       .subscribe { onZoomThresholdCrossed() }
 
-    adapter = LoiCardAdapter()
-    adapter.setLoiCardFocusedListener { mapFragment.setActiveLocationOfInterest(it) }
+    adapter = MapCardAdapter()
+    adapter.setLoiCardFocusedListener {
+      if (it?.loiId != null) {
+        mapFragment.setActiveLocationOfInterest(it.loiId)
+      }
+    }
     adapter.setCollectDataListener { navigateToDataCollectionFragment(it) }
-    mapContainerViewModel.loisWithinMapBoundsAtVisibleZoomLevel.observe(this) {
-      adapter.updateData(it)
+    mapContainerViewModel.loisWithinMapBoundsAtVisibleZoomLevel.observe(this) { lois ->
+      // TODO(#1541): Merge loi MapCardUiData with SuggestLoit MapCardUiData
+      adapter.updateData(
+        lois.map {
+          MapCardUiData(
+            name = it.caption ?: it.type.name,
+            subtitle = it.job.name,
+            surveyId = it.surveyId,
+            // TODO(#1483): Add submission count
+            submissionText = "No submissions",
+            loiId = it.id
+          )
```

Comment:
Can we move this logic to the VM?

Reply:
Done! I ended up reverting my changes to the LoiCardViewModel and updating the MapCardUiData to a sealed interface and wrapped the loi in an LoiCardUiData. Take a look and let me know what you think!

Reply:
Yes, this looks much better.